docs/styling/how-pagebuilder-styles-content.md

```diff
@@ -0,0 +1,77 @@
+
+```html
+<h2 data-content-type="heading"
+    data-appearance="default"
+    data-element="main"
+    data-pb-style="XDFNGK9">
+    My Heading
+</h2>
+```
+
+## Dynamic CSS selectors
+
+Page Builder generates its CSS selectors using one `id` and one `attribute`. This pattern is always the same, which gives all Page Builder CSS selectors a specificity of **110**. The anatomy of a Page Builder CSS selector is shown here:
+
+![Page Builder style selector](../images/pagebuilder-style-selector.svg)
+
+As noted, the CSS specificity of 110 is relatively low, which makes it possible to override these styles with your own custom CSS. For more information on CSS specificity and how it works, see https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity.
+
+## CSS rulesets
+
+To create the CSS rulesets for the selectors, Page Builder pulls the properties and values from the content type's form editor. The following example lists the CSS rules created from a Row content type (with minor changes made to the margins and paddings in the editor):
+
+```css
+#html-body [data-pb-style=WMWMCFQ] {
+    justify-content: flex-start;
+    display: flex;
+    flex-direction: column;
+    background-position: left top;
+    background-size: cover;
+    background-repeat: no-repeat;
+    background-attachment: scroll;
+    border-style: none;
+    border-width: 1px;
+    border-radius: 0;
+    margin: 0 0 10px;
+    padding: 10px;
+}
+````
+
+As end users change and save settings within a content type's editor, Page Builder updates the CSS ruleset to reflect those changes (and creates a new unique value for `data-pb-style`).
+
+![Page Builder content styling](../images/how-pagebuilder-styles-content.svg)
+
+## Inspecting Page Builder CSS
+
+If you inspect a page on your storefront built with Page Builder, you can see for yourself how Page Builder applies styles to its content types. For example, the following HTML is from a simple page with three content types: a Row, Heading, and Text. The highlighted parts show how the styles are defined and applied to Page Builder content types on a page.
+
+![Page Builder style selector](../images/pagebuilder-inspect-styling.png)
+
+1. **Html-body id**. The first thing to notice is the CSS `id=html-body` assigned to Magento storefront pages. Page Builder uses this `id` to construct all the CSS selectors it applies to its content types.
+
+2. **Internal style block**. Page Builder adds all the unique `data-pb-style` attribute styles (one for each content type on the page) to a single `<style>` block. This creates what's called an internal stylesheet for the page. In this example, the page contains three content types. So Page Builder has added three CSS rulesets to the `<style>` block, one for each content type on the page. If the page contained 10 content types, the `<style>` block would contain 10 CSS rulesets.
```

Comment:
I would be less specific about the count of rulesets because they are not aligned with a number of content types